CHARTER.md

```diff
@@ -0,0 +1,117 @@
+* New integration with supporting services such as secret stores.
+* Tooling to create and test new Crossplane extensions.
+* A conformance program for Crossplane extensions.
+
+As a framework, Crossplane:
+
+* Avoids building features specific to any particular compute plane in its core.
+  Instead features pertaining to any particular cloud provider, project, or tool
+  are enabled through extension points.
+* Focuses on APIs. Crossplane provides and recommends “golden path” tools to
+  build conformant extensions but does not require them. Anything with
+  conformant API schema and behavior may be considered a Crossplane extension -
+  regardless of how it is built. 
+* Optimizes for “vertical integration”. That is, Crossplane extensions are
+  intended and designed to be used in conjunction with Crossplane and each
+  other.
+* Prioritizes enabling a “separation of concerns” between control plane curators
+  and consumers. Curators deploy, extend, and configure control planes, while
+  consumers use them to orchestrate compute planes.
+
+## What Crossplane Is Not
+
+Crossplane is not intended to be a “batteries included” control plane for any
+particular use case. Instead it is a framework that you can deploy, configure,
+and extend to power control planes for your own highly bespoke use cases, often
+without writing any code.
+
+The project is a neutral place for vendors and individuals to come together in
+enabling control planes. To that extent, it does not police Crossplane
+extensions. Any extension that passes the appropriate Crossplane conformance
+suite may consider itself to be “a Crossplane extension” - regardless of how it
+is built, how it is licensed, who maintains it, how it is governed, or where it
+is hosted. Conformance suites are typically concerned with behavior and (where
+applicable) API shape, not internal implementation details.
+
+Concrete examples of things that are not in scope for the project include (but
+are not limited to):
+
+* Providing or mandating abstractions around specific use cases such as
+  provisioning applications (e.g. a PaaS API) or provisioning Kubernetes
+  clusters. 
+* Providing or mandating a configuration language (for example, CUE). Crossplane
+  provides a declarative API that may be targeted by arbitrary configuration
+  languages, and extension points that may be powered by arbitrary languages.
+* Enabling the use of Crossplane extensions without using Crossplane and vice
+  versa; this is not supported and may result in undefined behavior.
+* Providing or mandating opinions around control plane hosting and tenancy
+  models.
+* Offering ‘in-tree’ support for particular compute planes - e.g. AWS. All
+  features specific to any specific compute plane provider are built as
+  extensions.
+* Offering a package specification or manager for anything other than Crossplane
+  extensions.
+* Offering API clients such as command-line interfaces (CLIs) or web consoles.
```

Comment:
Does that mean Crossplane CLI (and [crank](https://github.com/crossplane/crossplane/tree/master/cmd/crank)) will be deprecated?

Reply:
Personally I lean toward grandfathering them in as a bit of an exception. I don't think we'd add any more functionality to them though.

Reply:
FWIW I would like to drop Crossplane CLI / crank. The most commonly used functionality is installing a package, and one could argue that it is an anti-pattern to use a CLI to imperatively carry out that operation (i.e. rather than applying a manifest).

Reply:
Sounds good. Maybe we can deprecate it around the time that we ratify this PR.

That said, do we have a good alternative for the creation of packages? That seems like something that should live in the Crossplane realm (it would be odd to be able to install them but not create them).

Reply:
I think the usual `docker build` or any other OCI image builder does the job, right? As long as the files exist in the right places.

Reply:
I'm going to resolve this conversation and leave it open ended for now. :) Personally I could go either way WRT grandfathering `crank` in or removing it - perhaps we can raise an issue to follow up on that once the charter is merged.